File: client/web/sudoo/lib/data/dto/auth/token_dto.dart
import 'package:sudoo/domain/model/auth/token.dart';

import '../../../domain/model/auth/role.dart';

class TokenDto {
  final String token;
  final String? refreshToken;
  final String? userId;
  final Role role;

  TokenDto({
    required this.token,
    required this.role,
    this.refreshToken,
    this.userId,
  });

  factory TokenDto.fromJson(Map<String, dynamic> json) {
    return TokenDto(
        token: json["token"],
        role: Role.fromValue(json["role"]),
        refreshToken: json["refreshToken"],
        userId: json["userId"]);
  }
}

extension Mapper on TokenDto {
  Token toToken() {
    return Token(token, refreshToken, role);
  }
}

File: client/mobile/Sudoo/app/src/main/java/com/sudo248/sudoo/data/dto/auth/TokenDto.kt
package com.sudo248.sudoo.data.dto.auth


/**
 * **Created by**
 *
 * @author *Sudo248*
 * @since 09:27 - 04/03/2023
 */
data class TokenDto(
    val token: String,
    val refreshToken: String? = null,
    val userId: String? = null,
)

